Preparations for move to new team

osfameron opened the pull request:
This includes
- script to create a mailmerge to advise people of change, and give an opt-out link
- an informational page about the transition...
- ...if called with opt-out link, then provide the option of a "delete my account" button
- minor tweaks to retirement banner and text

The deletion info is stored in a new table, and will require processing (either manual or automated TODO) to ensure that we only transfer the accounts that have not opted out.

zarino committed:
nicely styled opt-out pages

Note ordering of 'Gill Sans' typefaces to workaround
https://code.google.com/p/chromium/issues/detail?id=408093

osfameron committed:
Tweak wordings (inc retirement banner) for move

osfameron committed:
Script to create mailmerge for transfer

osfameron committed:
Draft T&C info and acceptance/rejection page

osfameron committed:
Update date of handover to 12th March

dracos commented on templates/tcs.tt:
```diff
@@ -0,0 +1,52 @@
+[% IF action %]
+
+    Thanks!  We've registered your intention to [% action %] your account and hassles
```
I believe there is an HTML injection here.

dracos commented on templates/tcs.tt:
```diff
@@ -0,0 +1,52 @@
+    <ul class="tc-promises">
+        <li><img src="/globe.png" width="64" height="64" alt="Globe"><br>HassleMe will<br>always be free</li>
+        <li><img src="/user_circle.png" width="64" height="64" alt="User Circle"><br>HassleMe will never<br>sell your email address,<br>or send you spam</li>
+        <li><img src="/heart.png" width="64" height="64" alt="Heart"><br>HassleMe will<br>be run by people<br>who love it.</li>
```
Please drop the content of the `alt` attributes – things for decoration should have `alt=""` :)

dracos commented on templates/tcs.tt:
```diff
@@ -0,0 +1,52 @@
+        <p>If, however, you’d prefer to not to be part of the new HassleMe, you can delete your HassleMe account and all associated Hassles:</p>
+
+        <p class="tc-decision__final">
+            <a class="tc-button" href="/TT/[% token %]/delete">Delete my HassleMe account</a>
```
Another one here.

dracos replied:
Ah, no, this one isn't, because token is checked.

dracos commented on templates/tcs.tt:
```diff
@@ -0,0 +1,52 @@
+            <a class="tc-button" href="/TT/[% token %]/delete">Delete my HassleMe account</a>
+        </p>
+    [% ELSE %]
+        <p>You should have received an email from us with a personalized link to be able to delete
```
"continue or delete your data" or similar? Don't want to imply deletion is all that can happen, I mean (even though I realise continuing is doing nothing.)

osfameron replied:
Hmm, though the backend does have the ability to register a "keep" message, that's been removed from the frontend, so the only way that the personalized link is different from the current page *is* the ability to delete your data.

dracos commented on web/hassleme.cgi:
```diff
@@ -412,6 +423,34 @@ EOF
+    my $token = $q->param('token');
+    my $email;
+
+    if (my $checked = check_token($q->param('token'))) {
```
You put token in `$token` a couple of lines above.

dracos commented on web/hassleme.cgi:
```diff
@@ -412,6 +423,34 @@ EOF
+
+    if (my $action = $q->param('action')) {
+        dbh()->do('insert into tc (email, action) values (?, ?)',
+                    {}, $email, $action);
```
This doesn't check that `$email` exists.

osfameron commented:
Thanks @dracos!

I'll review and correct later (probably tonight) and if all ok deploy tomorrow (pushing back release date to 10th March)

osfameron committed:
Fixes from PR

as per @dracos's comments

 - Fix HTML injection
 - Remove alt text for decorative images
 - Only run action if valid token & email provided
 - Clarify on no-email case that you don't need to do anything

osfameron commented:
Thanks, addressed issues hopefully, will squash and deploy tomorrow (lunch/afternoon) unless any more comments!
